streams/src/main/java/org/apache/kafka/streams/state/internals/WindowKeySchema.java

```diff
@@ -49,7 +43,21 @@ public long segmentTimestamp(final Bytes key) {
 
     @Override
     public HasNextCondition hasNextCondition(final Bytes binaryKey, final long from, final long to) {
-        return ITERATOR_HAS_NEXT;
+        return new HasNextCondition() {
+            @Override
+            public boolean hasNext(final KeyValueIterator<Bytes, ?> iterator) {
+                if (iterator.hasNext()) {
+                    final Bytes bytes = iterator.peekNextKey();
+                    final Bytes keyBytes = WindowStoreUtils.bytesKeyFromBinaryKey(bytes.get());
+                    if (!keyBytes.equals(binaryKey)) {
```

Comment:
I see, it is because the binary key becomes:
`ay\xFF\xFF\xFF\xFF\xFF\xFF\xFF\x00\x00\x00\x05`
which is greater than the previous one:
`aa\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x04`

So the lexicographic comparator will return  `aa\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x04` first, hence we stop iterating